src/py/flwr/server/superlink/linkstate/in_memory_linkstate.py

```diff
@@ -500,3 +501,13 @@ def acknowledge_ping(self, node_id: int, ping_interval: float) -> bool:
                 self.node_ids[node_id] = (time.time() + ping_interval, ping_interval)
                 return True
         return False
+
+    def get_serverapp_context(self, run_id: int) -> Optional[Context]:
+        """Get the context for the specified `run_id`."""
+        return self.contexts.get(run_id)
+
+    def set_serverapp_context(self, run_id: int, context: Context) -> None:
```

Comment:
worth renaming to `update_serverapp_context`? (to mimic `update_run_status`)

Reply:
Maybe `get/set_serverapp_context`? Because there might not exist a `Context` for the given run yet.